### PR TITLE
Issue #7091 Number Helper - Add Ordinal Number Suffix

### PR DESCRIPTION
**Pull request by ritesh-pandey**
This is simple function to add English ordinal number suffix after normal number. Function takes number as a parameter and returns number with suffix, like:
1st, 2nd, 3rd, 4th, 5th, 6th…

Added a function `ordinal` in `NumberHelper`

### CODE REVIEW

File: src/I18n/Number.php [hunk 1]
```diff
@@ -312,4 +313,18 @@ public static function formatter($options = [])
 
         return $formatter;
     }
+
+    /**
+     * Returns a formatted integer as an ordinal number string (e.g. 1st, 2nd, 3rd, 4th, [...])
+     *
+     * @param int|float $value An integer
```
Comment (dereuromark) [inline src/I18n/Number.php]: Why not just int?

**Comment by dereuromark** [comment]
Test cases are missing. Especially for the lib class.

File: src/I18n/Number.php (hunk 1 above)
Comment (ritesh-pandey) [inline src/I18n/Number.php]: Because `NumberFormatter::format` expects int/float.
http://php.net/manual/en/numberformatter.format.php

**Comment by ritesh-pandey** [comment]
@dereuromark I am figuring out/reading about how to write test cases.

**Comment by dereuromark** [comment]
Add the test cases to the corresponding files in /tests/ and run `php phpunit,phar /path/to/testfile` as documented on them.

File: src/I18n/Number.php [hunk 2]
```diff
@@ -312,4 +313,17 @@ public static function formatter($options = [])
...
+     * @return string
+     */
+    public static function ordinal($value)
+    {
+        $locale = Locale::getDefault();
```
Comment (markstory) [inline src/I18n/Number.php]: I think you want the current locale not the default locale.

**Comment by jhli** [comment]
Return 1., 2. etc for non english as by default - or rather: what happens if you - for instance - use french?

**Comment by dereuromark** [comment]
@jhli Yeah we should have at least 1 quite exotic test language to know what happens in some of those cases.

**Comment by ritesh-pandey** [comment]
@dereuromark Taking your "exotic test language" into account, I wrote some test cases.
Also, updated the Library function to read current locale instead of default locale.

**Comment by markstory** [comment]
Looks good to me.

**Comment by dereuromark** [comment]
:+1:

File: src/I18n/Number.php [hunk 3]
```diff
@@ -14,6 +14,7 @@
  */
 namespace Cake\I18n;
 
+use Cake\I18n\I18n;
```
Comment (antograssiot) [inline src/I18n/Number.php]: This is probably not useful as `Number` share the same namespace

Comment (ritesh-pandey) [inline src/I18n/Number.php]: @antograssiot Fixed :)

**Comment by lorenzo** [comment]
This needs to be added to the docs

**Comment by bcrowe** [comment]
Should we add an option to use `NumberFormatter:: SPELLOUT` as well?

**Comment by bcrowe** [comment]
Also... an option to set/choose the locale rather than just using current/default?

**Comment by lorenzo** [comment]
@bcrowe I would like the option to specify a locale

File: src/I18n/Number.php [hunk 4]
```diff
@@ -312,4 +312,17 @@ public static function formatter($options = [])
...
+    /**
+     * Returns a formatted integer as an ordinal number string (e.g. 1st, 2nd, 3rd, 4th, [...])
+     *
+     * @param int|float $value An integer
+     * @return string
+     */
+    public static function ordinal($value)
+    {
+        $locale = I18n::locale();
+        $formatter = new NumberFormatter($locale, NumberFormatter::ORDINAL);
```
Comment (ADmad) [inline src/I18n/Number.php]: Why create a new `NumberFormatter` instead of using `static::formatter()` like all other methods of the class do?

**Comment by ADmad** [comment]
> @bcrowe I would like the option to specify a locale

This would be solved if the `$options` argument is added to `ordinal()` and it uses `static::formatter($options)` to get formatter instance like other methods instead of creating a new formatter instance.

**Pull request merged**